src/addrdb.cpp

```diff
@@ -196,6 +196,15 @@ std::optional<bilingual_str> LoadAddrman(const std::vector<bool>& asmap, const A
         addrman = std::make_unique<AddrMan>(asmap, /* deterministic */ false, /* consistency_check_ratio */ check_addrman);
         LogPrintf("Creating peers.dat because the file was not found (%s)\n", path_addr);
         DumpPeerAddresses(args, *addrman);
+    } catch (const DbInconsistentError& e) {
+        // Addrman has shown a tendency to corrupt itself even with graceful shutdowns on known-good
+        // hardware. As the user would have to delete and recreate a new database regardless to cope
+        // with frequent corruption, we are restoring old behaviour that does the same, silently.
+        //
+        // TODO: Evaluate cause and fix, revert this change at some point.
```

Comment:
so far as 26909 is a fix for this issue - should we exclude 2nd commit from this PR? So, we can see if there's any more issues or that's no more

Reply:
It's been _marked_ as a fix but I'm not entirely sure if the bug has gone away. Initially this PR was going to only reintroduce silent overwrites but later figured that it be beneficial to reduce the _frequency_ of silent overwrites as well.

We could re-test the fix alone to validate it but as the errors were non-deterministic and we're approaching a release soon, I think bundling both to err on the side of caution is preferable.